license_sh/analyze/pipenv.py

```diff
@@ -0,0 +1,111 @@
+import subprocess
+import os
+import re
+import json
+import tempfile
+from anytree import AnyNode, PreOrderIter
+from license_sh.analyze.analyze_shared import run_askalono, LICENSE_GLOB
+from typing import Dict, List
+
+SUFFIX = ".dist-info"
+
+
+def get_analyze_pipenv_data(directory: str, tmpDir: str) -> Dict:
+    """Analyze pipenv dependencies
+
+  Args:
+      directory (str): Path to the project
+
+  Returns:
+      [Dict]: Analyzed data as dictionary
+  """
+    requirements_output = subprocess.run(
+        ["pipenv", "lock", "-r"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        cwd=directory,
+    ).stdout.decode("utf-8")
+    requirements_path = os.path.join(tmpDir, "requirements.txt")
+    with open(requirements_path, "w") as requirements_file:
+        requirements_file.write(requirements_output)
+
+    subprocess.run(
+        [
+            "pipenv",
+            "run",
+            "pip",
+            "install",
+            "-r",
+            requirements_path,
+            "--target",
+            tmpDir,
+        ],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        cwd=directory,
+    )
+    return run_askalono(tmpDir)
+
+
+def get_pipenv_analyze_dict(directory: str) -> Dict:
+    """Get pipenv analyze dictionary
+
+	Args:
+		directory (str): Path to the project
+
+	Returns:
+		Dict: Dependency {name}-{version} as key, license text and analyzed license name and value
+	"""
+    data_dict = {}
+    with tempfile.TemporaryDirectory() as dirpath:
+        license_data = get_analyze_pipenv_data(directory, dirpath)
+        for item in license_data:
+            dep_id = item.get("path").split("/")[-2].split(SUFFIX)[0]
```

Comment:
I'd appreciate a comment about a data structure here or maybe reasonably spliiting this into more variables. Right now I see that you get a `path` attribute and split it and take -2's index. 

Please use unpacking more frequently. https://stackabuse.com/unpacking-in-python-beyond-parallel-assignment/

Code is more readable then accessing directly to an array with constants.

Reply:
Sure, added.... You are right, looks much better